src/arrays/programs_involving_data_sequences/hard/q7/MainTest.java

```diff
@@ -0,0 +1,102 @@
+package arrays.programs_involving_data_sequences.hard.q7;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.utils.ArrayUtil.*;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public static int n = 10;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = arrayToInput(generateRandomArray(-10, 10, n));
+        return new Clause[]{
+                new StringLiteral("Enter an array of " + n + " integers to be squared:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("The squared array is:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder(),
+                new Optional(new StringLiteral(" "))
+        };
+    }
+
+    public void runMain() {
+        SquareUp.main(new String[0]);
+    }
+
+    public static int[] sq(int[] arr) {
+        int[] input = arr.clone();
+        for (int i = 0; i < input.length; i++)
+            input[i] = (int) Math.pow(input[i], 2);
+        return input;
```

Comment:
To change this to be different than the solution code, you can have input[i]*=input[i] instead of Math.pow